client/state/automated-transfer/enhancer.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Internal dependencies
+ */
+import { isAutomatedTransferActive } from 'state/selectors/';
+import {
+	AUTOMATED_TRANSFER_ELIGIBILITY_REQUEST,
+	AUTOMATED_TRANSFER_ELIGIBILITY_UPDATE,
+	AUTOMATED_TRANSFER_STATUS_SET,
+	THEME_TRANSFER_INITIATE_FAILURE,
+	THEME_TRANSFER_INITIATE_PROGRESS,
+	THEME_TRANSFER_INITIATE_REQUEST,
+	THEME_TRANSFER_INITIATE_SUCCESS,
+	THEME_TRANSFER_STATUS_FAILURE,
+	THEME_TRANSFER_STATUS_RECEIVE
+} from 'state/action-types';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+const WHITELIST = [
+	AUTOMATED_TRANSFER_ELIGIBILITY_REQUEST,
+	AUTOMATED_TRANSFER_ELIGIBILITY_UPDATE,
+	AUTOMATED_TRANSFER_STATUS_SET,
+	THEME_TRANSFER_INITIATE_FAILURE,
+	THEME_TRANSFER_INITIATE_PROGRESS,
+	THEME_TRANSFER_INITIATE_REQUEST,
+	THEME_TRANSFER_INITIATE_SUCCESS,
+	THEME_TRANSFER_STATUS_FAILURE,
+	THEME_TRANSFER_STATUS_RECEIVE
+];
+
+/**
+ * Redux store enhancer that updates dispatch to ignore actions when an
+ * automated transfer is in progress.
+ *
+ * @param  {Function} createStore Original store creator
+ * @return {Function}             Modified store creator
+ */
+export const automatedTransferEnhancer = createStore => ( reducer, initialState, enhancer ) => {
+	const store = createStore( reducer, initialState, enhancer );
+
```

Comment:
can we also put the environment check here? it should only run once so the extra overhead won't be significant. if we're not in development or have the right config flag we can just return `store` instead of augmenting it. this will make sure that even if we deploy on accident it won't impact the runtime of Calypso